src/app/api/auth/[...nextauth]/route.ts

```diff
@@ -2,208 +2,8 @@
-        const email = profile.email;
-        const existingUser = await getSubscriberByEmail(email);
-
-        if (existingUser) {
-          token.subscriber = existingUser;
-          if (account.access_token && account.refresh_token) {
-            const updatedUser = await updateFxAData(
-              existingUser,
-              account.access_token,
-              account.refresh_token,
-              JSON.stringify(profile)
-            );
-            token.subscriber = updatedUser;
-          }
-        }
-        if (!existingUser && email) {
-          const verifiedSubscriber = await addSubscriber(
-            email,
-            profile.locale,
-            account.access_token,
-            account.refresh_token,
-            JSON.stringify(profile)
-          );
-          token.subscriber = verifiedSubscriber;
-
-          const allBreaches = await getBreaches();
-          const unsafeBreachesForEmail = await getBreachesForEmail(
-            getSha1(email),
-            allBreaches,
-            true
-          );
-
-          // Send report email
-          const utmCampaignId = "report";
-          const l10n = getL10n();
-          const subject = unsafeBreachesForEmail?.length
-            ? l10n.getString("email-subject-found-breaches")
-            : l10n.getString("email-subject-no-breaches");
-
-          const breachLogos = await getBreachIcons(allBreaches);
-          const data = {
-            breachedEmail: email,
-            breachLogos: breachLogos,
-            ctaHref: getEmailCtaHref(utmCampaignId, "dashboard-cta"),
-            heading: "email-breach-summary",
-            recipientEmail: email,
-            subscriberId: verifiedSubscriber,
-            unsafeBreachesForEmail,
-            utmCampaign: utmCampaignId,
-          };
-          const emailTemplate = getTemplate(
-            data,
-            signupReportEmailPartial,
-            l10n
-          );
-
-          await initEmail(process.env.SMTP_URL);
-          await sendEmail(data.recipientEmail, subject, emailTemplate);
-        }
-      }
-      return token;
-    },
-    async session({ session, token }) {
-      if (token.fxa) {
-        session.user.fxa = {
-          locale: token.fxa.locale,
-          twoFactorAuthentication: token.fxa.twoFactorAuthentication,
-          metricsEnabled: token.fxa.metricsEnabled,
-          avatar: token.fxa.avatar,
-          avatarDefault: token.fxa.avatarDefault,
-          subscriptions: token.fxa.subscriptions,
-        };
-      }
-      if (token.subscriber) {
-        session.user.subscriber = token.subscriber;
-      }
-      return session;
-    },
-  },
-  events: {
-    signIn(message) {
-      log.debug("fxa-confirmed-profile-data", message.user);
-    },
-    signOut(message) {
-      log.debug("logout", message.token.email ?? undefined);
-    },
-  },
-};
+import NextAuth from "next-auth";
+import { authOptions } from "../../utils/auth"
```

Comment:
So it seems like the test failure and error just really are what they say:
> Type error: Route "src/app/api/auth/[...nextauth]/route.ts" does not match the required types of a Next.js Route. "authOptions" is not a valid Route export field.

Moving `authOptions` into a separate file and not exporting it directly from the route resolves the issue. Not sure why this only started to be an issue now and not in recent PRs.

Reply:
Really weird that that didn't turn up before! /cc @rhelmer for potential merge conflicts in case his auth work also modifies `authOptions`.